docs/source/portfolio_optimisation/hierarchical_risk_parity.rst

```diff
@@ -39,6 +80,41 @@ Implementation
         .. automethod:: __init__
 
 
+.. note::
+    |h4| Using Custom Input |h4_|
+    We provide great flexibility to the users in terms of the input data - they can either pass their own pre-calculated input
+    matrices/dataframes or leave it to us to calculate them. A quick reference on common input parameters which you will encounter
+    throughout the portfolio optimisation module:
+        * :py:mod:`asset_prices`: Dataframe/matrix of historical raw asset prices **indexed by date**.
+        * :py:mod:`asset_returns`: Dataframe/matrix of historical asset returns. This will be a :math:`TxN` matrix where :math:`T` is the time-series and :math:`N` refers to the number of assets in the portfolio.
+        * :py:mod:`expected_asset_returns`: List of expected returns per asset i.e. the mean of historical asset returns. This refers to the parameter :math:`\mu` used in portfolio optimisation literature. For a portfolio of 5 assets, ``expected_asset_returns = [0.45, 0.56, 0.89, 1.34, 2.4]``.
+        * :py:mod:`covariance_matrix`: The covariance matrix of asset returns.
+
+
+.. tip::
+    |h4| Using Custom Distance Matrix |h4_|
+    The hierarchical clustering step in the algorithm uses a distance matrix to calculate the clusters and form the hierarchical
+    tree. By default, we use the distance matrix mentioned in the original paper,
+
+    .. math::
+        D(i, j) = \sqrt{\frac{1}{2} * (1 - \rho(i, j))}
+
+    Here, :math:`\rho` refers to the correlation matrix of assets. Users can specify their own custom matrix to be used instead of
+    the default one by passing an :math:`NxN` symmetric pandas dataframe or a numpy matrix using the :py:mod:`distance_matrix`
+    parameter. **Note that the** :py:mod:`covariance_matrix` **is still requested for computing the clustered variances.**
+
+    |h4| Constructing a Long/Short Portfolio |h4_|
+    By default, the weights are allocated assuming a long portfolio i.e. all the weights are positive. However, users can also opt
+    to have specific assets in their portfolio shorted, by specifying a custom list through the :py:mod:`side_weights` parameter.
+    For e.g. in a portfolio of 5 assets, if you want the 3rd and the 5th asset to be shorted, then simply pass a list containing
+    1s and -1s for long and short positions respectively. In the above case, ``side_weights = [1,1,-1,1,-1]``, where you
+    have -1 at the 3rd and 5th index. By default, a list of 1s will be initialised if no custom input is specified.
+
+    |h4| Different Linkage Methods |h4_|
+    HRP, by default, uses the single-linkage clustering algorithm. (See the tip under the HCAA algorithm for more details.)
```

Comment:
Perhaps, worth adding that the list of supported methods is `...` . And later - They're described in the tip under the HCAA algorithm.